### PR TITLE
docs: fix claims to roles description in access controls reference

### LINKED ISSUE #25026: [v.12.x] /docs/pages/access-controls/reference.mdx
## Applies To
https://goteleport.com/docs/access-controls/reference/

## Details
The part with `claims_to_roles` is confusing:

```yaml
      # the `claims_to_roles` mapping works the same as it does in
      # the OIDC connector, with the added benefit that the roles being mapped to
      # can also be matchers. the below mapping says that users with
      # the claims `groups: admins` can request any role in the system.
      claims_to_roles:
        - claim: 'projects'
          # matches all group names with a leading 'product-'
          value: '^product-(.*)$'
          # generates a role name from the value capture
          roles: ['$1-admin']
```

Description mentions `groups:admins` and any role, the code refers to `projects:product-*` and `*-admin` role.

### PR DESCRIPTION
**Pull request by stevenGravy**
Fixes #25026

The description didn't match the claims to roles example.

### CODE REVIEW

File: docs/pages/includes/role-spec.mdx [hunk 1]
```diff
@@ -236,7 +236,9 @@ spec:
       # the `claims_to_roles` mapping works the same as it does in
       # the OIDC connector, with the added benefit that the roles being mapped to
       # can also be matchers. the below mapping says that users with
-      # the claims `groups: admins` can request any role in the system.
+      # claims that match to `projects: project-(.*)` can request
+      # roles that match with '$1-admin' with the first match group.
```
Comment (zmb3) [inline docs/pages/includes/role-spec.mdx:240]: Is there a difference between "match to" and "match with"? 

If not, I would use consistent language. 

Probably also worth mentioning that these are regular expressions too.

Comment (stevenGravy) [reply docs/pages/includes/role-spec.mdx:241]: thanks, update

File: docs/pages/includes/role-spec.mdx [hunk 2]
```diff
@@ -235,8 +235,11 @@ spec:
 
       # the `claims_to_roles` mapping works the same as it does in
       # the OIDC connector, with the added benefit that the roles being mapped to
-      # can also be matchers. the below mapping says that users with
-      # the claims `groups: admins` can request any role in the system.
+      # can also be matchers. this example leverages Teleport's regular expression
+      # support that allows for dynamic mapping from claims. the below mapping
+      # says that users with claims that match with `projects: project-(.*)` can request
+      # roles that match with '$1-admin' with the first match group.
```
Comment (ptgott) [inline docs/pages/includes/role-spec.mdx:241]:
```suggestion
      # can also be matchers. 
      #
      # This example leverages Teleport's regular expression support, which allows 
      # for dynamic mapping from claims. The below mapping says that users with
      # claims that match "projects: project-(.*)" can request roles that match 
      # "$1-admin", where "$1" is the first capture group in the 
      # regular expression.
```

File: docs/pages/includes/role-spec.mdx [hunk 3]
```diff
@@ -235,8 +235,14 @@ spec:
...
-      # can also be matchers. the below mapping says that users with
-      # the claims `groups: admins` can request any role in the system.
+      # can also be matchers. 
+      #
+      # This example leverages Teleport's regular expression support, which allows 
+      # for dynamic mapping from claims. The below mapping says that users with
+      # claims that match "projects: project-(.*)" can request roles that match 
+      # "$1-admin", where "$1" is the first capture group in the 
+      # regular expression.
+      # Example: `projects: project-foo` claim allows requesting `foo-admin` role
```
Comment (ptgott) [inline docs/pages/includes/role-spec.mdx:245]:
```suggestion
      # Example: the "projects: project-foo" claim allows a user to request the
      # "foo-admin" role
```

**Review by ptgott — APPROVED** [review]
Approved with a minor readability suggestion

File: docs/pages/includes/role-spec.mdx [hunk 4]
```diff
@@ -235,8 +235,15 @@ spec:
...
-      # can also be matchers. the below mapping says that users with
-      # the claims `groups: admins` can request any role in the system.
+      # can also be matchers. 
+      #
+      # This example leverages Teleport's regular expression support, which allows 
+      # for dynamic mapping from claims. The below mapping says that users with
+      # claims that match "projects: project-(.*)" can request roles that match 
+      # "$1-admin", where "$1" is the first capture group in the 
+      # regular expression.
+      # Example: the "projects: project-foo" claim allows a user to request the
```
Comment (alexfornuto) [inline docs/pages/includes/role-spec.mdx:245]: To match the value below, would it be:

```suggestion
      # Example: the "projects: product-foo" claim allows a user to request the
```
?

Comment (stevenGravy) [reply docs/pages/includes/role-spec.mdx]: Thanks @alexfornuto , yes corrected.

Comment (stevenGravy) [reply docs/pages/includes/role-spec.mdx]: Please approve if that's good to go @alexfornuto

**Review by alexfornuto — APPROVED** [review]

**Pull request merged**